File: lingting-spring-jackson/src/test/java/live/lingting/spring/jackson/TestObjectMapperCustomizer.java
package live.lingting.spring.jackson;

import com.fasterxml.jackson.databind.ObjectMapper;
import lombok.Getter;
import org.springframework.stereotype.Component;

/**
 * @author lingting 2024-02-02 15:59
 */
@Getter
@Component
public class TestObjectMapperCustomizer implements ObjectMapperCustomizer {

	private ObjectMapper objectMapper;

	@Override
	public ObjectMapper apply(ObjectMapper mapper) {
		objectMapper = mapper.copy();
		return objectMapper;
	}

}

File: lingting-spring-jackson/src/test/kotlin/live/lingting/spring/jackson/TestObjectMapperCustomizer.kt
package live.lingting.spring.jackson

import com.fasterxml.jackson.databind.ObjectMapper
import org.springframework.stereotype.Component

/**
 * @author lingting 2024-02-02 15:59
 */
@Component
class TestObjectMapperCustomizer : ObjectMapperCustomizer {
    var objectMapper: ObjectMapper? = null
        private set

    override fun apply(mapper: ObjectMapper) {
        objectMapper = mapper.copy()
    }
}
